internal/cmd/common.go

```diff
@@ -0,0 +1,110 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package cmd
+
+import (
+	"github.com/edgexfoundry/edgex-cli/internal/config"
+	"github.com/edgexfoundry/edgex-cli/internal/format"
+	"github.com/spf13/cobra"
+)
+
+var verbose, metadata, data, command, notification, scheduler, json bool
+
+func getSelectedService() string {
+	if metadata {
+		return "core-metadata"
+	} else if data {
+		return "core-data"
+	} else if command {
+		return "core-command"
+	} else if notification {
+		return "core-notification"
+	} else if scheduler {
+		return "core-scheduler"
+	} else {
+		return ""
+	}
+}
+
+func dataService(cmd *cobra.Command, api string, name string) error {
+	s := config.GetCoreService("core-data")
+	value, _, _, err := format.GetServiceValue(s, api, name)
+	if err != nil {
+		cmd.Println(err.Error())
+	} else {
+		cmd.Println(value)
+	}
+	return nil
+}
+
+func standardService(cmd *cobra.Command, api string, name string) error {
+	serviceName := getSelectedService()
+
+	if verbose {
+		if serviceName != "" {
+			s := config.GetCoreService(serviceName)
+			_, jsonValue, url, err := format.GetServiceValue(s, api, name)
+			if err != nil {
+				cmd.Println(err.Error())
+			} else {
+				cmd.Printf("%s: %s: %s\n", serviceName, url, jsonValue)
+			}
+		} else {
+			format.PrintServiceTable(api, format.Debug, "")
+		}
+	} else if json {
+		if serviceName == "" {
+			serviceName = "core-data"
+		}
+		s := config.GetCoreService(serviceName)
+		_, jsonStr, _, err := format.GetServiceValue(s, api, "")
+		if err != nil {
+			cmd.Println(err.Error())
+		} else {
+			cmd.Println(jsonStr)
+		}
+	} else {
+		if serviceName != "" {
+			s := config.GetCoreService(serviceName)
+			value, _, _, err := format.GetServiceValue(s, api, name)
+			if err != nil {
+				cmd.Println(err.Error())
+			} else {
+				cmd.Printf("%s: %s\n", serviceName, value)
+			}
+		} else {
+			format.PrintServiceTable(api, format.Value, name)
+		}
+	}
+	return nil
+}
+
+func addFormatFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVarP(&json, "json", "j", false, "get JSON reply")
+	cmd.Flags().BoolVarP(&verbose, "debug", "d", false, "show debug output")
+
+}
+
+func addStandardFlags(cmd *cobra.Command) {
+	addFormatFlags(cmd)
+	cmd.Flags().BoolVarP(&data, "data", "", false, "use core-data service endpoint")
+	cmd.Flags().BoolVarP(&command, "command", "c", false, "use core-command service endpoint")
+	cmd.Flags().BoolVarP(&metadata, "metadata", "m", false, "use core-metadata service endpoint")
+	cmd.Flags().BoolVarP(&scheduler, "scheduler", "s", false, "use core-scheduler service endpoint")
```

Comment:
```suggestion
	cmd.Flags().BoolVarP(&scheduler, "scheduler", "s", false, "use support-scheduler service endpoint")
```

Reply:
oops, absolutely, copy-and-paste error there in the naming of those services. well spotted. thanks. fixed.